webpack.config.js

```diff
@@ -188,5 +188,8 @@ module.exports = {
 		] ),
 		new DependencyExtractionWebpackPlugin( { injectPolyfill: true } ),
 	],
+	watchOptions: {
+		ignored: '!packages/**/(build|build-module)/**/*',
```

Comment:
I don't think we ever care about the `build` folder in the context of webpack (I might be wrong). However, there are a few packages that don't have `build-module` folder generated, example: `@wordpress/is-shallow-equal`. I don't think we have a pattern for ES5-based packages but we could enforce using `lib` subfolder to make it all work here. We could add linter check for `package.json` files that ensures that when `module` is not defined then `main` has to include `lib/index.js` to make it all work consistently so we could have:

```suggestion
		ignored: '!packages/**/(lib|build-module)/**/*',
```

The allowed values for `main` and `module` in `package.json` would be:

```json
{
    "main": "lib/index.js"
}
```

or 

```json
{
    "main": "build/index.js",
    "module": "build-module/index.js"
}
```

What do you think?

Reply:
https://github.com/WordPress/gutenberg/blob/a429d461afca6d1c457dec255e040d80e6cd348f/webpack.config.js#L111-L112

What about `build-styles`? Should it be also excluded from the list of ignored patterns?

Reply:
> I don't think we ever care about the `build` folder in the context of webpack (I might be wrong).

Hm, you might be right about this. At least for our setup, any package with a `build` would expect to have a `build-module`, and Webpack should always be deferring to `build-module` over `build`.

You're right this should be (and isn't currently) including those non-transpiled modules. And you're also right that our conventions here aren't very consistent. There was actually some conversation about this at https://github.com/WordPress/gutenberg/pull/18942#issuecomment-601772189 . I think we should want this consistency, and then we could include `lib` like you mention.

Reply:
> What about `build-styles`? Should it be also excluded from the list of ignored patterns?

Ah, I had thought about this, and didn't think that Webpack was doing anything with the CSS. Clearly it is, so yes, it should be considered in the whitelist pattern.